File: articles/stakeholder-values-and-decision-legitimacy/python/run_all_stakeholder_legitimacy_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Stakeholder Values and Decision Legitimacy."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "stakeholder_mapping.py",
    "value_weight_model.py",
    "burden_analysis.py",
    "procedural_legitimacy.py",
    "threshold_checks.py",
    "legitimacy_index.py",
    "decision_record_exporter.py",
    "stakeholder_values_decision_legitimacy_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python stakeholder legitimacy workflows completed.")


if __name__ == "__main__":
    main()

File: articles/stakeholder-values-and-decision-legitimacy/r/run_all_stakeholder_legitimacy_workflows.R
# run_all_stakeholder_legitimacy_workflows.R
# Run all R workflows for Stakeholder Values and Decision Legitimacy.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "stakeholder_values_decision_legitimacy_workflow.R",
  "stakeholder_scores.R",
  "burden_tables.R",
  "procedural_legitimacy_tables.R",
  "threshold_review_tables.R",
  "legitimacy_summary.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R stakeholder legitimacy workflows completed.")
